src/bp/nlu-core/entities/microsoft-extractor/microsoft-extractor.test.ts

```diff
@@ -0,0 +1,263 @@
+import 'bluebird-global'
+import _ from 'lodash'
+import path from 'path'
+import { unlinkSync } from 'fs'
+import { MicrosoftEntityExtractor } from '.'
+import { SystemEntityCacheManager } from '../entity-cache-manager'
+
+describe('Microsoft Extract Multiple', () => {
+  let microsoft: MicrosoftEntityExtractor
+  let testCachePath = path.join(process.APP_DATA_PATH || '', 'cache', 'testCache.json')
+  beforeAll(() => {
+    const duckCache = new SystemEntityCacheManager(testCachePath, true)
+    microsoft = new MicrosoftEntityExtractor(duckCache)
+  })
+
+  beforeEach(async () => {
+    microsoft.resetCache()
+  })
+
+  afterAll(() => {
+    unlinkSync(testCachePath)
+  })
+
+  test('Return nothing for unsupported lang', async () => {
+    const examples = ['один два три четыре пять', 'Я говорю по русски сегодня, но и завтра вечером']
+    const res = await microsoft.extractMultiple(examples, 'ru')
+    res.forEach(r => {
+      expect(r).toEqual([])
+    })
+  })
+
+  // Note we could add numbers as global recognizers as well.
+  test('Return phone number, ip address, mention, hashtag, email, url for unsupported lang', async () => {
+    const results = [
+      [
+        {
+          confidence: 1,
```

Comment:
So if microsoft returns the same payload with confidence: 0.99 the test will fail?

In this test, you assert that returned payloads are these **exact** one, but you don't have to. 

The name of your test is "Return phone number, ip address, mention, hashtag, email, url for unsupported lang".

So all you have to check is if those entities where extracted even if language is not supported.

```ts
const expected = ["email", "number", ...]
const actual= await microsoft.extractMultiple(examples, 'ru')
for (const ([x, a]) of _.zip(expected, actual)) {
    expect(x.type).toBe(a)
}
```
something like this above is more than enough to check those types are extracted even when language is not supported.

Reply:
Confidence is indeed hardcoded in the extractor... 

The point that I don't want the recognizer to give me each number of the ip address extracted "by error", like 135 then 19 then 84 then 102 and also 135.19.84.102 is an ip address.  I understand that it's quite a broad test but I don;t know if I want to change it...